gpflow/functions.py

```diff
@@ -110,7 +110,15 @@ def __init__(self, A: TensorType = None, b: TensorType = None) -> None:
         MeanFunction.__init__(self)
         A = np.ones((1, 1), dtype=default_float()) if A is None else A
         b = np.zeros(1, dtype=default_float()) if b is None else b
-        self.A = Parameter(np.atleast_2d(A))
+        if isinstance(A, Parameter):
+            if len(A._shape) >= 2:
+                self.A = A
+            else:
+                raise TypeError(
```

Comment:
This should probably be a ValueError not a TypeError:

```suggestion
                raise ValueError(
```